ConcurrentAPI.adoc

```diff
@@ -0,0 +1,244 @@
+== Distributed Concurrent API
+
+=== Motivation
+
+Coming from an initial issue where we need to provide an implementation for vert.x cluster manager for the shared data feature,
+provide concurrent and distributed programming capabilities.
+
+* Distributed Locks
```

Comment:
We had a discussion a while ago about what should be called distributed and what should we call clustered.
For future reference:
* distributed: information which is divided/sharded among the nodes in the cluster.
* clustered: information which is global to the cluster (owned by a single node or by all nodes)